### DISCUSSION TITLE
Any tips on achieving active ragdoll pose matching?

### DISCUSSION THREAD

**Discussion by schemesmith** [godot-jolt/godot-jolt#843] [Questions]
Hello

I'm looking for any suggestions on how to get the active ragdoll behavior shown in this video (which is using the jolt physics engine)
https://www.youtube.com/watch?v=lYHhe6HLbs4

With the default physics engine I'm able to achieve this more or less with setting angular limits for each PhysicalBone (Using Generic6DOF) in a skeleton to target angles, but it seems in Jolt the ragdoll joints don't really enforce the angular limits as well and the ragdoll just crumples into a puddle.

Any advice or direction on where to look?

Thank you!

**Comment by mihe** [comment]
Not really, unfortunately. Messing with the limits like that won't work well with the way I've implemented the joints, as you've found out already.

The way this is ideally meant to be done in Jolt is by using so called position motors, which are motors that drive joints to a specific position/orientation rather than to a specific velocity (like regular motors). However, the only way you have access to these through this extension is by using 6DOF joints, enabling linear/angular springs and then changing the `equilibrium_point` property for each respective axis. The joints will obviously be springy though, but I suppose you can crank up stiffness/frequency to a high value.

(Note that the rotation order for the angular equilibrium point is `EULER_ORDER_ZYX` rather than the default `EULER_ORDER_YXZ`.)

**Reply by schemesmith** [reply]
Oh hey yea that works pretty well thank you!

Also this is out of my wheelhouse but would this be the best way to recreate the SkeletonMapper test from Jolt?
https://github.com/jrouwe/JoltPhysics/blob/master/Samples/Tests/Rig/SkeletonMapperTest.cpp

**Reply by mihe** [reply]
`equilibrium_point` could satisfy the `DriveToPoseUsingMotors` part you see there, but the rest is obviously extraneous.

Keep in mind that Jolt's built-in ragdolls (which I don't expose) also [does things](https://github.com/jrouwe/JoltPhysics/blob/56adf54375d4393be7752acd796e1ca3ea3f5474/Jolt/Physics/Ragdoll/Ragdoll.cpp#L51) to help prevent the ragdoll from exploding. So without doing something similar you might not get the kinds of results you see in Jolt's demo clips.

**Reply by schemesmith** [reply]
ah got it that makes sense - thank you!
